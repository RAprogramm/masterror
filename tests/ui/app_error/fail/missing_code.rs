// SPDX-FileCopyrightText: 2025 RAprogramm <andrey.rozanov.vl@gmail.com>
//
// SPDX-License-Identifier: MIT

use masterror::{AppCode, AppErrorKind, Error};

#[derive(Debug, Error)]
enum MissingCode {
    #[error("with code")]
    #[app_error(kind = AppErrorKind::NotFound, code = AppCode::NotFound)]
    WithCode,
    #[error("without code")]
    #[app_error(kind = AppErrorKind::Service)]
    WithoutCode,
}

fn main() {}
