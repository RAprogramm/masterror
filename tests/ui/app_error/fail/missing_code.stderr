error: AppCode conversion requires `code = ...` in #[app_error(...)]
 --> tests/ui/app_error/fail/missing_code.rs:9:5
  |
9 |     #[app_error(kind = AppErrorKind::Service)]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused imports: `AppCode` and `AppErrorKind`
 --> tests/ui/app_error/fail/missing_code.rs:1:17
  |
1 | use masterror::{AppCode, AppErrorKind, Error};
  |                 ^^^^^^^  ^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` (part of `#[warn(unused)]`) on by default
